src/client/particles.cpp

```diff
@@ -620,14 +620,20 @@ const core::aabbox3df &ParticleBuffer::getBoundingBox() const
 		return m_mesh_buffer->BoundingBox;
 
 	core::aabbox3df box;
+	bool            need_first_vertex = true;
 	for (u16 i = 0; i < m_count; i++) {
 		// check if this index is used
 		static_assert(quad_indices[1] != 0);
 		if (m_mesh_buffer->getIndices()[6 * i + 1] == 0)
 			continue;
 
 		for (u16 j = 0; j < 4; j++)
-			box.addInternalPoint(m_mesh_buffer->getPosition(i * 4 + j));
+			if (need_first_vertex) {
+				box.reset(m_mesh_buffer->getPosition(i * 4 + j));
+				need_first_vertex = false;
+			} else {
+				box.addInternalPoint(m_mesh_buffer->getPosition(i * 4 + j));
+			}
 	}
 
 	m_mesh_buffer->BoundingBox = box;
```

Comment:
no practical consequences but the box will be wrong if `m_count == 0`

Reply:
I believe that was also the case prior to this change. The same issue is present in a number of places in irr/ as well, so I'm planning to repair all of them at once separately.

Reply:
#15507 will make those fixes a lot nicer because the default constructor will do the right thing.

Reply:
I disagree with merging a partial fix for this problem now.

Reply:
I can go ahead and fix all the usages in this PR.